components/core/src/reducer/Pipeline.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef REDUCER_PIPELINE_HPP
+#define REDUCER_PIPELINE_HPP
+
+#include <memory>
+#include <set>
+#include <vector>
+
+#include "GroupTags.hpp"
+#include "Operator.hpp"
+#include "Record.hpp"
+#include "RecordGroup.hpp"
+
+namespace reducer {
+enum class PipelineInputMode : uint8_t {
+    IntraStage,
+    InterStage
```

Comment:
What is an InterStage pipeline?

Reply:
A pipeline that calls push_interstage on the first operator instead of push_intrastage